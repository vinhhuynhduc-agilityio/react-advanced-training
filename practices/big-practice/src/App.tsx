import React from "react";

// components
import Header from "./components/Header";

const App: React.FC = () => {
  return (
    <div className="bg-slate-100 h-screen w-screen">
      <Header />
    </div>
  );
};

export default App;
