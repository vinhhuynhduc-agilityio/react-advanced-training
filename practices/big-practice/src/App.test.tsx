import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "./App";

// Mock Dashboard component
jest.mock("./pages/Dashboard/Dashboard", () => ({
  ...jest.requireActual("./pages/Dashboard/Dashboard"),
  default: () => <div data-testid="dashboard-component">Mocked Dashboard</div>,
}));

describe("App Component", () => {
  it("matches snapshot for App", () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
