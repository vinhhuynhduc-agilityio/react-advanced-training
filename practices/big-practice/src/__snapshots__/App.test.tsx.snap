// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `<div />`;

exports[`App Component matches snapshot for App 1`] = `
<div>
  <div
    data-testid="dashboard-component"
  >
    Mocked Dashboard
  </div>
</div>
`;
